server.go

```diff
@@ -33,13 +33,16 @@ func (p *PasswordService) Authenticate(request *Packet) (*Packet, error) {
 }
 
 // NewServer return a new Server given a addr, secret, and service
-func NewServer(addr string, secret string, service Service) *Server {
+func NewServer(addr string, secret string, service Service, clients ...*Clients) *Server {
```

Comment:
There is too many arguments in this function.
You can define a new function with a NewServerRequest struct to pass all those arguments and keep backward compatible